### PR TITLE
moved anti keybinds from eh to built in keybind introduced in arma3 v…

### LINKED ISSUE #1760: Battle menu under custom key
**Is your feature request related to a problem? Please describe.**
Battle menu button is hardcoded as a battle menu invoker. It is problematic because some of people use it to something else.

**Describe the solution you'd like**
Any button should be possible

**Describe alternatives you've considered**
none

**Additional context**
Possible change in fn_keys.sqf, line 6

`if (inputAction "User16" > 0) then
	{...}`

Then any button under User16 may be used. Of course it may be any free UserXX.
I already tested it and it works also with artillery combination.
It may also require info about binding need in init customHint, but this is rather easy to solve.

### PR DESCRIPTION
**Pull request by HakonRydland**
…2.06

## What type of PR is this.
1. [ ] Bug
2. [x] Change
3. [ ] Enhancement

### What have you changed and why?
Information:
    since arma 3 v2.06 and the move to mod, we are now able to have custom keybinds similare to vanilla keybinds of arma, this pr removes the old key eh and replaces them with the new native keybind system.
    per https://community.bistudio.com/wiki/Arma_3:_Modded_Keybinding

### Please specify which Issue this PR Resolves.
closes #1760

### Please verify the following and ensure all checks are completed.

1. [x] Have you loaded the mission in LAN host?
2. [ ] Have you loaded the mission on a dedicated server?

### Is further testing or are further changes required?
1. [x] No
2. [ ] Yes (Please provide further detail below.)

### How can the changes be tested?
Steps: 
1) load into mission
2) test keybinds
3) open configure -> controls -> antistasi
and modify the keybinds there
4) test with new keybinds
5) reset keybinds to default
7) check that you cant place vehicles while battleMenu is open (only thing blocked currently)
6) modify keybind, unload mod, launch, the relaunch with mod loaded again, then check keybinds stored properly

********************************************************
Notes:

### CODE REVIEW

File: A3A/addons/core/Stringtable.xml [hunk 1]
```diff
@@ -2922,12 +2922,12 @@
         <Russian>Информационная строка</Russian>
       </Key>
       <Key ID="STR_antistasi_dialogs_toggle_info_bar_body_off">
-        <Original>You've turned off top information bar. To enable it, press ALT+HOME.</Original>
-        <Russian>Вы выключили отображение информационной строки. Чтобы включить её снова, нажмите ALT+HOME.</Russian>
+        <Original>You've turned off top information bar. To enable it, press </Original>
```
Comment (Bob-Murphy) [inline A3A/addons/core/Stringtable.xml:2925]: We can't simply delete the button and leave half a sentence.

Comment (HakonRydland) [reply A3A/addons/core/Stringtable.xml:2925]: constructed from localized string + button name

Comment (HakonRydland) [reply A3A/addons/core/Stringtable.xml:2925]: although i havent tried with unbound key, but this hint would never be shown while the key is unbound

Comment (Bob-Murphy) [reply A3A/addons/core/Stringtable.xml:2925]: I got it. I'm used to having all information in the string and not being it partially code and added to the A3A_fnc_customHint at a different place.
Makes sense.

**Comment by Bob-Murphy** [comment]
Is there a reason you left in `A3A/addons/core/functions/Base/fn_keys.sqf` as well as commented out references to it?

File: A3A/addons/core/config.cpp [hunk 2]
```diff
@@ -37,3 +37,5 @@ class A3A {
 
 #include "defines.hpp"
 #include "dialogs.hpp"
+
```
Comment (Bob-Murphy) [inline A3A/addons/core/config.cpp:40]: unnecessary empty line?

File: A3A/addons/core/keybinds/CfgUserActions.hpp [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    };
+
+    class GVAR(infoBar) {
+        displayName = "Toggle info bar";
+        tooltip = "Toggle visability of Antistasis information bar at the top of the screen";
+        onActivate = ACTION(infoBar);
+    };
+
+    class GVAR(earPluggs) {
+        displayName = "Toggle ear pluggs";
```
Comment (Bob-Murphy) [inline A3A/addons/core/keybinds/CfgUserActions.hpp:23]: it's `plugs`

File: A3A/addons/core/keybinds/CfgUserActions.hpp [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+        tooltip = "Toggle use of ear pluggs (no effect when using ace)";
```
Comment (Bob-Murphy) [inline A3A/addons/core/keybinds/CfgUserActions.hpp:24]: it's `plugs`

File: A3A/addons/core/keybinds/CfgUserActions.hpp [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+        onActivate = ACTION(artyMenu);
+    };
+
+    class GVAR(infoBar) {
+        displayName = "Toggle info bar";
+        tooltip = "Toggle visability of Antistasis information bar at the top of the screen";
+        onActivate = ACTION(infoBar);
+    };
+
+    class GVAR(earPluggs) {
```
Comment (Bob-Murphy) [inline A3A/addons/core/keybinds/CfgUserActions.hpp:22]: it's `earPlugs`

File: A3A/addons/core/keybinds/CfgUserActions.hpp [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+        displayName = "Toggle ear pluggs";
+        tooltip = "Toggle use of ear pluggs (no effect when using ace)";
+        onActivate = ACTION(earPluggs);
```
Comment (Bob-Murphy) [inline A3A/addons/core/keybinds/CfgUserActions.hpp:25]: it's `earPlugs`

File: A3A/addons/core/keybinds/UserActionGroups.hpp [hunk 7]
```diff
@@ -0,0 +1,12 @@
+class UserActionGroups {
+    class PREFIX {
+        name = "Antistasi";
+        isAddon = 0;
+        group[] = {
+            QGVAR(battleMenu),
+            QGVAR(artyMenu),
+            QGVAR(infoBar),
+            QGVAR(earPluggs)
```
Comment (Bob-Murphy) [inline A3A/addons/core/keybinds/UserActionGroups.hpp:9]: it's `earPlugs`

File: A3A/addons/core/keybinds/UserActionsConflictGroups.hpp [hunk 8]
```diff
@@ -0,0 +1,14 @@
+class UserActionsConflictsGroups {
+    class ActionGroups {
+        PREFIX[] = {
+            QGVAR(battleMenu),
+            QGVAR(artyMenu),
+            QGVAR(infoBar),
+            QGVAR(earPluggs)
```
Comment (Bob-Murphy) [inline A3A/addons/core/keybinds/UserActionsConflictGroups.hpp:7]: it's `earPlugs`

File: A3A/addons/core/keybinds/fn_keyActions.sqf [hunk 9]
```diff
@@ -0,0 +1,57 @@
...
+        if (ctrlShown _infoBarControl) then {
+            ["KEYS", true] call A3A_fnc_disableInfoBar;
+            [localize "STR_antistasi_dialogs_toggle_info_bar_title", localize "STR_antistasi_dialogs_toggle_info_bar_body_off" + " " + _keyName, false] call A3A_fnc_customHint;
+        } else {
+            ["KEYS", false] call A3A_fnc_disableInfoBar;
+            [localize "STR_antistasi_dialogs_toggle_info_bar_title", localize "STR_antistasi_dialogs_toggle_info_bar_body_on" + " " + _keyName, false] call A3A_fnc_customHint;
+        };
+    };
+
+    case QGVAR(earPluggs): {
```
Comment (Bob-Murphy) [inline A3A/addons/core/keybinds/fn_keyActions.sqf:42]: it's `earPlugs`

File: A3A/addons/core/keybinds/CfgDefaultKeysPresets.hpp [hunk 10]
```diff
@@ -0,0 +1,11 @@
+#include "\a3\ui_f\hpp\definedikcodes.inc"
+class CfgDefaultKeysPresets {
+    class Arma2 {
+        class Mappings {
+            GVAR(battleMenu)[] = {DIK_Y};
+            GVAR(artyMenu)[] = {0x2A130015}; //combo Left shift + Y (no double tap)
+            GVAR(infoBar)[] = {0x381300C7}; //combo Left alt + Home (no double tap)
+            GVAR(earPluggs)[] = {DIK_END};
```
Comment (Bob-Murphy) [inline A3A/addons/core/keybinds/CfgDefaultKeysPresets.hpp:8]: it's `earPlugs`

File: A3A/addons/core/keybinds/CfgUserActions.hpp [hunk 11]
```diff
@@ -0,0 +1,28 @@
...
+
+    class GVAR(artyMenu) {
+        displayName = "Artillery menu";
+        tooltip = "Open Antistasis artillery support menu (commander only)";
+        onActivate = ACTION(artyMenu);
+    };
+
+    class GVAR(infoBar) {
+        displayName = "Toggle info bar";
+        tooltip = "Toggle visability of Antistasis information bar at the top of the screen";
```
Comment (Bob-Murphy) [inline A3A/addons/core/keybinds/CfgUserActions.hpp:18]: it's `visibility`

**Review by Bob-Murphy — CHANGES_REQUESTED** [review]
Typos

File: A3A/addons/core/config.cpp (hunk 2 above)
Comment (HakonRydland) [reply A3A/addons/core/config.cpp:40]: its to separate grouped includes, defines.hpp is required for dialogs.hpp

File: A3A/addons/core/keybinds/CfgUserActions.hpp [hunk 12]
```diff
@@ -15,14 +15,14 @@ class CfgUserActions {
...
     };
 
-    class GVAR(earPluggs) {
-        displayName = "Toggle ear pluggs";
-        tooltip = "Toggle use of ear pluggs (no effect when using ace)";
-        onActivate = ACTION(earPluggs);
+    class GVAR(earPlugs) {
+        displayName = "Toggle ear plugs";
+        tooltip = "Toggle use of ear plugs (no effect when using ace)";
+        onActivate = ACTION(earplugs);
```
Comment (Bob-Murphy) [inline A3A/addons/core/keybinds/CfgUserActions.hpp:25]: I think the P must be uppercase?

**Review by Bob-Murphy — APPROVED** [review]
I reviewed the P.
I approve the P.

**Review by CalebSerafin — CHANGES_REQUESTED** [review]
Okay, the change is quite good.
The only blocking change is that the stringtable entries should use formating entities like %1. More info in that issue.

**Pull request merged**